ksql-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -261,15 +262,19 @@ private static boolean isTopicDeleteEnabled(final AdminClient adminClient) {
           String.valueOf(nodes.iterator().next().id())
       );
 
+      ;
+
       final Map<ConfigResource, Config> config = ExecutorUtil.executeWithRetries(
           () -> adminClient.describeConfigs(Collections.singleton(resource)).all().get(),
           ExecutorUtil.RetryBehaviour.ON_RETRYABLE);
 
       return config.get(resource)
           .entries()
           .stream()
-          .anyMatch(configEntry -> configEntry.name().equalsIgnoreCase("delete.topic.enable")
-              && configEntry.value().equalsIgnoreCase("true"));
+          .filter(configEntry -> configEntry.name().equalsIgnoreCase("delete.topic.enable"))
+          .findFirst()
+          .map(configEntry -> configEntry.value().equalsIgnoreCase("true"))
+          .orElse(Defaults.DeleteTopicEnable());
```

Comment:
Which default is this? I think the default changed in Kafka in version 1.0 so if its the current default this breaks compatibility with pre-1.0

Reply:
I think it makes sense to default to true if the broker doesn't return the config.

Reply:
It would be nice to add a comment in the code for why we assume this default: ie. we assume this when the config is filtered out of the response, as in the case for confluent cloud servers, in which case assuming a default of `true` is fine (since that is the default on the servers).